### PR TITLE
feature: Add goneovim Cask

### PR DESCRIPTION
**Pull request by elken**
Add a Cask for https://github.com/akiyosi/goneovim

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/goneovim.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+cask "goneovim" do
+  version "0.4.11"
+
+  if Hardware::CPU.in_rosetta2? || Hardware::CPU.arm?
+    sha256 "8d9d08db2f96e32944f41c9e7de29de83581deab607b199b3a1516cffcc234e4"
+    url "https://github.com/akiyosi/goneovim/releases/download/v#{version}/Goneovim-#{version}-macos11.tar.bz2"
+    app "Goneovim-#{version}-macos11/goneovim.app"
```
Comment (carlocab) [inline Casks/goneovim.rb:7]: This seems like a Big Sur build, and not a native ARM one:
```
❯ file Goneovim-0.4.11-macos11/goneovim.app/Contents/MacOS/goneovim
Goneovim-0.4.11-macos11/goneovim.app/Contents/MacOS/goneovim: Mach-O 64-bit executable x86_64
```
We should probably be checking for the macOS version here, and not whether we're on an M1 mac.

File: Casks/goneovim.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+cask "goneovim" do
```
Comment (carlocab) [inline Casks/goneovim.rb:1]: Shouldn't this have a dependency on the `neovim` formula?

File: Casks/goneovim.rb (hunk 1 above)
Comment (elken) [reply Casks/goneovim.rb:7]: Good point, it was linked to an issue mentioning Silicon so assumed it was a "two birds one stone" build, will amend

File: Casks/goneovim.rb (hunk 2 above)
Comment (elken) [reply Casks/goneovim.rb:1]: Not sure how I missed that! My bad

File: Casks/goneovim.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  if MacOS.version <= :big_sur
+    sha256 "8d9d08db2f96e32944f41c9e7de29de83581deab607b199b3a1516cffcc234e4"
+    url "https://github.com/akiyosi/goneovim/releases/download/v#{version}/Goneovim-#{version}-macos11.tar.bz2"
+    app "Goneovim-#{version}-macos11/goneovim.app"
+  else
+    sha256 "b33f1ab2038424263fd0d49058b16d43ce3eabd56a603c20b24aaed24295f051"
+    url "https://github.com/akiyosi/goneovim/releases/download/v#{version}/Goneovim-#{version}-macos.tar.bz2"
+    app "Goneovim-#{version}-macos/goneovim.app"
+  end
```
Comment (carlocab) [inline Casks/goneovim.rb:12]:
```suggestion
  if MacOS.version < :big_sur
    sha256 "b33f1ab2038424263fd0d49058b16d43ce3eabd56a603c20b24aaed24295f051"
    url "https://github.com/akiyosi/goneovim/releases/download/v#{version}/Goneovim-#{version}-macos.tar.bz2"
    app "Goneovim-#{version}-macos/goneovim.app"
  else
    sha256 "8d9d08db2f96e32944f41c9e7de29de83581deab607b199b3a1516cffcc234e4"
    url "https://github.com/akiyosi/goneovim/releases/download/v#{version}/Goneovim-#{version}-macos11.tar.bz2"
    app "Goneovim-#{version}-macos11/goneovim.app"
  end
```
`MacOS.version <= :big_sur` is always true (unless you're running the beta macOS 12, in which case you get the one built for versions earlier than Big Sur, which isn't what you want).

Comment (cho-m) [reply Casks/goneovim.rb:12]: Probably should match other casks and use `<= :catalina`

Comment (carlocab) [reply Casks/goneovim.rb:12]: Suggestion revised.

**Comment by elken** [comment]
Should I also add a caveat about the package being unsigned?

**Comment by miccal** [comment]
> Should I also add a caveat about the package being unsigned?

No, that is not necessary, thanks.

**Pull request merged**